frontend/src/metabase/collections/containers/CollectionContent.jsx

```diff
@@ -150,8 +165,25 @@ function CollectionContent({ collection, collectionId, isAdmin, isRoot }) {
                 onCopy={handleCopy}
               />
 
-              <Search.ListLoader query={unpinnedQuery} wrapped>
-                {({ list: unpinnedItems, metadata }) => {
+              <Search.ListLoader
+                query={unpinnedQuery}
+                loadingAndErrorWrapper={false}
+                keepPreviousList
+                wrapped
+              >
+                {({
+                  list: unpinnedItemsList = [],
+                  previousList: previousUnpinnedItemsList = [],
+                  metadata = {},
+                  loading: loadingUnpinnedItems,
+                }) => {
+                  // Prevents table UI flashing while performing a GET request
+                  // (e.g. for pagination or sorting)
+                  const unpinnedItems =
+                    unpinnedItemsList.length > 0
+                      ? unpinnedItemsList
+                      : previousUnpinnedItemsList;
```

Comment:
What happens when a list is refetched with an added filter or something and the latest query (with filter applied) resolves with an empty array. In that scenario we'd want to show no results but we wouldn't know it here.

Reply:
I'm not sure if that is a problem here, but it could potentially be a problem in another place that uses this sort of logic.

Reply:
Great catch! My case only covers pagination and sorting, so I missed that. We definitely need to rely on the loading state here as well. The previous version should be used only if there is an ongoing request